components/ColonyQuestions.js

```diff
@@ -15,7 +16,31 @@ const StyledQuestionLabel = styled.label`
   text-align: center;
 `;
 
-export default function ColonyQuestions() {
+export default function ColonyQuestions({ currentColony }) {
+  const [formData, setFormData] = useState({});
+
+  useEffect(() => {
+    const storedFormData = localStorage.getItem(
+      `colonyFormData_${currentColony.id}`
+    );
+    if (storedFormData) {
+      setFormData(JSON.parse(storedFormData));
+    }
+  }, [currentColony]);
+
+  function handleInputChange(event) {
+    const { name, value, type, checked } = event.target;
+    const newValue = type === "checkbox" ? checked : value;
+    setFormData((prevFormData) => {
+      const newFormData = { ...prevFormData, [name]: newValue };
+      localStorage.setItem(
+        `colonyFormData_${currentColony.id}`,
+        JSON.stringify(newFormData)
+      );
+      return newFormData;
+    });
+  }
```

Comment:
I'm spotting some `getItem` and `setItem` usage here that leads me to believe that you may have chosen a localStorage implementation that we don't usually recommend. We usually recommend using the [use-local-storage-state](https://github.com/astoilkov/use-local-storage-state) package, which you can implement similarly to this:

```javascript
// import { useState } from "react";
import useLocalStorageState from "use-local-storage-state";

function Counter() {
  // const [count, setCount] = useState(0);
  const [count, setCount] = useLocalStorageState("count", { defaultValue: 0 });

  return (
    <div>
      <p>Count: {count}</p>
      <button onClick={() => setCount(count + 1)}>Increment</button>
    </div>
  );
}
```